rest_framework/serializers.py

```diff
@@ -892,19 +892,23 @@ def create(self, validated_data):
         # Save many-to-many relationships after the instance is created.
         if many_to_many:
             for field_name, value in many_to_many.items():
-                setattr(instance, field_name, value)
+                set_many(instance, field_name, value)
 
         return instance
 
     def update(self, instance, validated_data):
         raise_errors_on_nested_writes('update', self, validated_data)
+        info = model_meta.get_field_info(instance)
```

Comment:
@tomchristie - I'm still hesitant about the extra call to `get_field_info` here. How performance intensive is it to redo all of the field introspection?